### PR TITLE
Change: Split Game options into General, Graphics, and Sound tabs.

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

Game Options is very large and has lots of different settings on it.

image

## Description

This change splits the Game Options window into three small tabs, "General", "Base sets" and "Video":

image
image
image

There isn't really a specific widget that could be used to represent tabs (these came in later games...) so this uses a standard button in a contrasting colour.

"Interface size" is left on the (default) General tab, as in modern terminology it fits more into a "graphics" option than a "video" option, but I don't think a separate Graphics tab is appropriate.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Comment by nielsmh** [comment]
Oh, I was about to start doing this myself too.
Personally I'd call the tabs General, Graphics, Sound, and put the grahics baseset selection on the Graphics tab.

**Comment by nielsmh** [comment]
Oh, and we kind of do have tabs in the GUI: Station Details and Vehicle Details windows have buttons on the bottom to switch the lower pane of the window.

**Comment by PeterN** [comment]
Yes, that's true, but they're not very prominent either (I'd forgotten about them as tabs) so maybe not the best example to follow.

**Comment by nielsmh** [comment]
Definitely, the volume sliders in the current layout were just jammed in there to add _some_ way for players to adjust things from the intro screen. It was an easy place to put them without altering the layout too much.

**Comment by 2TallTyler** [comment]
I second @nielsmh's suggestion. General/Graphics/Sound is a pretty universal organization scheme these days, so players know where to find what they want. `Base sets` is very OpenTTD-specific.

For bonus points cleaning up the bug tracker...does this close #10170? 😃

**Comment by PeterN** [comment]
Potentially, more so with the requested tab changes.

**Comment by andythenorth** [comment]
Does this have the bonus effect that the window won't be cut off on some screen sizes (with no way to resize)?

**Comment by LordAro** [comment]
Yellow on grey doesn't look great... Any other options?

**Comment by PeterN** [comment]
> Does this have the bonus effect that the window won't be cut off on some screen sizes (with no way to resize)?

That's what #10170 referenced above is.

> Yellow on grey doesn't look great... Any other options?

It's orange. Grey on grey doesn't really work either.

**Comment by PeterN** [comment]
Tweaked the order... and these are now yellow rather than orange buttons, I dunno...

Not convinced I've got the volume sliders laid out well, but they're definitely more prominent.

image
image
image

**Comment by LordAro** [comment]
Since you're in the area...

I can't find it anywhere, but I definitely recall an issue where teh game options window was too wide due to the Video driver string. Splitting it within the string is probably too difficult, but perhaps it could be put on its own line, under "Current driver: "? Would help a little

**Comment by 2TallTyler** [comment]
`Sound/Music` could just be `Sound`, I think. I usually see music included in the Sound tab in other games.

No opinion on orange vs yellow tabs.

**Comment by PeterN** [comment]
Yellow looks better I think, but perhaps the extra panel around them could go. Need to play :)

**Comment by LC-Zorg** [comment]
I think I like this change. :) I don't know, I'm not sure. From my point of view, it is not really needed, the problem of too large a window can also be solved with a slider, but the advantage I see in such a layout may be the ability to add or move other options that could now fit better here. One of the options that would be worth considering would be the option to choose a font.
The yellow color is ok in my opinion. I like the updated division, it's similar to the one I was thinking to myself about (of course I won't add anything, it's just tiresome sketches again;). What I don't really like is the stretched bars in the general settings tab - these lines are a bit tiring. Might be worth shortening it? The second reservation concerns the position of the language setting. For me it seems to be the most important of all the settings. This is not an insurmountable problem, but if a player accidentally or out of curiosity sets a language completely unfamiliar to him, e.g. Arabic or Hebrew, he may have a bit of a problem with finding the setting. You can also think about adding a language selection icon (wikipedia and many other sites), which I think is quite widely known.

A bit off-topic at the end, but a bit not. I admit that I also thought (for a long time) about some changes in terms of settings in the game. One of the things that probably gnawed at me the most was the question: why are these settings divided into two windows? This makes some sense, but not always. **Maybe it would be worth moving all interface settings to the options window and leaving only the gameplay settings in the settings window?**
Anyway, I had another long-developing concept of combining all settings in one window. Interestingly, there are also the General, Graphics and Sound tabs here. :) In addition, there is also the Gameplay and Interface tab. I tried to think about it more comprehensively and the changes are related to other windows as well. I don't know if it's good, but maybe you should consider another arrangement?
This is what it looks like to me...
X++
1. Minimized main menu with a vertical layout inspired by the idea that TrueBrain presented some time ago. Add-ons, new games and settings tabs have been reduced to single buttons.
2. The settings window contains all game settings, including the "weird" ones that are not present today (in general, I don't think the game has too many settings, the impression of their excess is rather the result of not the best division of them), as well as keyboard shortcuts and, additionally, settings for online games regarding both the player and the server. The latter could be very useful for easier establishment, configuration and maintenance of online games. Today, setting up a server is something reserved for a rather small number of players, because it requires proper knowledge. I think it's worth simplifying it.
Changed color, buttons, details and assigned options are irrelevant here. It's mainly about the general concept of dividing all the settings in the game into five tabs.
3. One window for all add-ons. It is based on my earlier slightly developed concept. The GS and AI selection layout could remain almost unchanged, including the ability to save and select presets, which is currently not possible. Such an arrangement would also make sense in the case of GS if they could have dedicated functions (complex or only: conditions for the development of cities / conditions for the development of industry / taxes / restrictions / rewards / ...) and could be combined with each other.
4. One window to start a new game, both on random map, heightmap and scenario. The idea is a combination of the concept of adding presets by TrueBrain and the old concept of andythenorth, which assumed the possibility of selecting all settings for a new game in one window. The scenario editor removed from the main menu would be under the scenario tab. What you can see is the first finished sketch.

**Comment by PeterN** [comment]
Shuffled things a little, including moving Language to the first general item.

image
image
image

I think the combined settings/options window needs thinking about, perhaps as a separate tab as what you have there is way more complex and overwhelming.

File: src/settings_gui.cpp [hunk 1]
```diff
@@ -163,6 +163,18 @@ static const std::map<int, StringID> _scale_labels = {
 	{  500, STR_GAME_OPTIONS_GUI_SCALE_5X },
 };
 
+static const std::map<int, StringID> _volume_labels = {
```
Comment (LordAro) [inline src/settings_gui.cpp:166]: Why is this necessary now, when it was not previously? Does it need some doxyment?

**Review by LordAro — APPROVED** [review]
Excellent stuff. Ship it.

File: src/settings_gui.cpp (hunk 1 above)
Comment (PeterN) [reply src/settings_gui.cpp:166]: To be honest it's not absolutely needed, but having the labels makes the two sliders look better (IMHO) when placed directly together.
